surveyor.py

```diff
@@ -5,477 +5,457 @@
+
+        try:
+            self._results_collector = [header] #A dd header to results collector.
+            
+            # Query
+            if query:
+                # if a query is specified run it directly
+                log_echo(f"Running Custom Query: {query}", self._log)
+                if raw:
+                    raw_results = product.process_search(Tag('query'), base_query, query) # Returns data directly from API
+
+                    if self._edr in ['cbr', 'cbc']: 
+                        if len(raw_results) > 0:
+                            log_echo(f"\033[92m-->query: {len(raw_results)} results \033[0m", self._log, use_tqdm=self._use_tqdm)
+                        else:
+                            log_echo(f"-->query: {len(raw_results)} results", self._log, use_tqdm=False)
+                        return raw_results
+                    
+                    elif self._edr in ['cortex', 'dfe', 's1']:
+                        self._results_collector = [] 
+                        for tag, results in product.get_results().items():
+                            self._results_collector.append(results)                    
+                            if len(self._results_collector ) > 0:
+                                log_echo(f"\033[92m-->query: {len(self._results_collector )} results \033[0m", self._log, use_tqdm=self._use_tqdm)
+                            else:
+                                log_echo(f"-->query: {len(self._results_collector )} results", self._log, use_tqdm=self._use_tqdm)
+                                
+                    if len(self._results_collector) > 1:
+                         return self._results_collector
+                    else:
+                        return "No results"
+                        
+                else:
+                    product.process_search(Tag('query'), base_query, query)
+                
+                for tag, results in product.get_results().items():
+                    self._write_results(results, query, "query", tag)
+            
+            # IOC
+            if ioc_list and (ioc_source and ioc_type):
+                # run search based on IOC file
 
+                    log_echo(f"Processing IOC list: {ioc_source}", self._log)
+                    ioc_list = [x.strip() for x in ioc_list]
+
+                    product.nested_process_search(Tag(f"IOC - {ioc_source}", data=ioc_source), {ioc_type: ioc_list}, base_query)
 
-table_template: Tuple[int, int, int, int, int, int] = (30, 30, 30, 30, 30, 30)
-table_template_str = f'{{:<{table_template[0]}}} ' \
-                     f'{{:<{table_template[1]}}} ' \
-                     f'{{:<{table_template[2]}}} ' \
-                     f'{{:<{table_template[3]}}}'
+                    for tag, results in product.get_results().items():
+                        self._write_results(results, ioc_source, 'ioc', tag)
+
+            # Sigma
+            sigma_rules = list()
```

Comment:
This line overwrites the existing `sigma_rules` variable to be an empty list. Is that expected?